### PR TITLE
Reworks Night Vision, Light Drinker, and Hyper-Sensitive to be actually interesting

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

V5gg97IUzH

Reworks Night Vision, Light Drinker, and Hyper-Sensitive to be actually interesting.

Light Drinker now makes you black out drunk immediately upon metabolizing booze. Light Drinker means Light Drinker.

Hyper-Sensitive now only magnifies bad moodlets, and taking bad moodlets will immediately negatively impact your sanity.
As a side effect of the refactor of moodlet modifiers for this, PTSD has been modified to affect bad moodlets as well.
Moodlet modifier calculation is on a per-moodlet basis instead of after all of the moodlets have been tallied together.

Night Vision is now Nightmare Vision.

## Why It's Good For The Game

image (96)
These are some of the most used quirks in the game for this specific combination, and all 3 of them are incredibly boring, being:
1 minor debuff that's easily avoidable(don't drink)
1 buff that's easily usable (take a shower, have massive mood buffs for ages)
1 massive buff that's permanent (night vision so darkness is completely meaningless to you)

These are all boring effects and against what we want from Quirk design so I've replaced them all with more interesting variants as described above.

The visual effect for Nightmare Vision will be changed if we're able to get it working; the goal is to do pixelated darkness that's hard to make out the contents of.

## Changelog
:cl:
balance: Reworks Night Vision, Light Drinker, and Hyper-Sensitive to be actually interesting.
balance: Light Drinker now makes you black out drunk immediately upon metabolizing booze. Light Drinker means Light Drinker.
balance: Hyper-Sensitive now only magnifies bad moodlets, and taking bad moodlets will immediately negatively impact your sanity.
balance: As a side effect of the refactor of moodlet modifiers for this, PTSD has been modified to affect bad moodlets as well.
balance: Moodlet modifier calculation is on a per-moodlet basis instead of after all of the moodlets have been tallied together.
balance: Night Vision is now Nightmare Vision.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
I do not like the nightvision = hellvision, both because I tuned current nightvision (potentially poorly I admit) and I think the idea is fine, but also because it detracts from hellvision as a concept
also none in their right might would ever want that as a permanent effect so it fails as a positive quirk

**Comment by necromanceranne** [comment]
I dared someone to use nightmare vision goggles the whole round and apparently it gave him pretty bad eyestrain, so maybe don't let people do that.

**Comment by zoomachina** [comment]
you can't have light drinker be a -2 with that kind of effect

**Comment by Colovorat** [comment]
>Light Drinker now makes you black out drunk immediately upon metabolizing booze. Light Drinker means Light Drinker.

Goof never had a booze in his life

>Night Vision is now Nightmare Vision

that should give points now instead of removing


incredibly bad changes that were given 0 thought

File: code/datums/mood.dm [hunk 1]
```diff
@@ -209,18 +212,23 @@
...
 
 	for(var/category in mood_events)
 		var/datum/mood_event/the_event = mood_events[category]
+		var/mood_change = the_event.mood_change
+		if(mood_change > 0)
+			mood_change *= mood_modifier[MOOD_MODIFIER_GOOD]
+		else
+			something_bad_happened = TRUE
+			mood_change *= mood_modifier[MOOD_MODIFIER_BAD]
 		mood += the_event.mood_change
```
Comment (necromanceranne) [inline code/datums/mood.dm:229]: I think this is supposed to use your new var and not the event's value, right?

**Comment by RedBaronFlyer** [comment]
this is somehow nearly as bad as the previous PR

**Comment by KickPaw** [comment]
Bad PR, that deletes features. We don't have many negative quirks that couldn't completely kill you. Also night vision is a bit useless unless you use mesons or thermals, so to use it you sacrifice your glasses slot.

**Comment by Yttriums** [comment]
Please stop posting that superset chart with raw counts of perk selections like it's meaningful. It's offensive to those of us that understand data analysis.

Please consider that nerfing common negative perks will ruin the perk builds of most people using the perk system whatsoever, not just those who are 'powergaming' night vision.

At the very least, atomize what are clearly 3 distinct changes.

**Comment by jughu** [comment]
to be fair the night vision one is deserved since thats just a gameplay buff since most antags can move trough darkness to suprise the enemy

**Comment by ATH1909** [comment]
I think the reason that light drinker has stayed untouched for as long as it has been is that it's actually kind of nice to have 2 "free" points available to buy some of the milder quirks with.

Maybe we should add those 2 points to your base quirk point total if we nerf light drinker?

**Comment by Admiral-Gigaton** [comment]
Light drinker instant blackoutmakes it so no one will ever take it, as a single unit will mean some guy gets to grief you as you can only watch as he tosses all your shifts work into the SM

At the least make it give way more quirk points, at the most make it a whole new quirk

**Comment by Formerxboxgamer** [comment]
1 unit of any booze to make someone unable to play the game sounds awful for balance. Stop trying to butcher these perks just because they’re picked a lot.

**Comment by Deadgebert** [comment]
> me getting ready to chem dart 1u of booze into known light drinkers

**Comment by Iamgoofball** [comment]
@LemonInTheDark as stated in the OP, the nightmare vision visual effect is slated for changes

File: code/datums/mood.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/datums/mood.dm:229]: whoops, good catch

**Comment by KingkumaArt** [comment]
So many negative quirks are either:
- minus a bunch of points so it's not worth taking unless you're loading up on one of the high cost "good" quirks, (that are often of questionable utility)
- minus a small amount of points but still way too much of a pain in the ass to take as anything other than a purposeful handicap when heirloom exists
- some variation on physical injury and with the exception of wheelchair and rds none of them impact roleplay or interaction other than role-playing as a person in a waiting room

I'd love some more low cost negative quirks akin to food allergy etc because half the people taking these quirks take them because nearly every other negative quirk is stupidly high impact and they only want something like musician

**Comment by RedBaronFlyer** [comment]
Looking at this with a fresh pair of eyes, here's my thoughts:

If night vision is really broken at the moment then tweaking how good it is is in order, but not a glass table slam nerf where no one uses it after the change (AKA the Valve TF2 balancing method)

Light drinker seems entirely fine and I fail to see the issue. This seems like an overnerf for no tangible reason.

Hypersensitive is perhaps a bit too good sometimes but it's balanced around being a mood glass cannon, it is easy to keep your mood high but you can very easily have your mood torpedoed and it's hard to recover. If it **had** to be changed, it should be a neutral quirk but retain it's current mechanics.

IMO hypersensitive, light drinker, and heirloom are all symptoms of the issue where most of the other quirks are too high impact to the point that less people want to run them. Most positive quirks are _incredibly_ low impact though. It seems like night vision is the only potential broken positive quirk at the moment, and I don't think the solution is your "glass table slam" method.

**Comment by Iamgoofball** [comment]
i think i will tweak the good quirks to be high impact and thus worth fucking yourself up for

**Comment by jughu** [comment]
NV quirk got buffed unintendedly when it got reworked it doesn't define your character so this should be either everyone gets soft NV or nobody should get it it shouldnt be a quirk

**Comment by nianjiilical** [comment]
i know youve said you're tweaking nightmare vision but please for the love of christ do not put it on a quirk **until** you do because people are going to take this quirk not understanding what it is and then ahelp to get it removed 99.9% of the time

**Comment by SomeoneYouProbablyKnow** [comment]
Light drinker – that’s really funny. Might want to change the name to something more descriptive like “Rowdy drunk”. Question, with the lower alcohol intake needed is the blackout personality gonna have better control over your character, since you are not at the stage where everything’s spinning and you can barely stand? Or is that rolled up in the blackout itself? Might be an issue, might be nothing. Also yea -2 is a bit too low, no one is gonna take it to min max so might as well reward the people taking it for the comedy of it.

Also, it’d be nice to have a 0 point quirk that does what old light drinker did. I had it on because slurred speech is funny but carrying industrial amounts of alcohol and killing your liver for a joke is annoying.

Night vision – yea fuck night vision, ass quirk, never used for anything but powergaming. No clue what to do with it instead, though. Better vision if you stand still? Like eyes acclimating or something? Or just give the feature to lizards 🦎, maint cretura 

Hypersensitive – don’t like it, this iteration feels way more like a challenge mode quirk. The old one wasn’t even the problem, only its cost (and it fucked you over often enough). Could just bump it to 4 cost and see how it plays out, doubt people are gonna be investing points into it.

>i think i will tweak the good quirks to be high impact and thus worth fucking yourself up for

That’s how you give MORE power to meta loadouts, right now someone powergaming quirks nets you slightly higher movement speed and a meson buff. You make those quirks more powerful without reworking ALL the negative quirks to be in line you’ll have more experienced players starting with noticeably stronger characters simply because they know the meta loadout.

I think strong quirks should come with inbuilt downsides not just their point value (You want night vision? Enjoy being light sensitive as well or something like that). Because if you can mix and match and there’s something worth mixing and matching for it WILL be abused.

**Comment by optimumtact** [comment]
the whole point of the quirks system is to not be high impact.

**Comment by optimumtact** [comment]
>1 massive buff that's permanent (night vision so darkness is completely meaningless to you)

night vision is only 3 tiles isn't it? it's slightly bigger than the normal radius but not the full view range.

**Comment by Iamgoofball** [comment]
@optimumtact someone reworked it to be screen-wide, heres how it looks rn
dreamseeker_8pSVzoaQOi

**Comment by optimumtact** [comment]
I dunno if you're joking but that is a picture of you seeing nothing.

**Comment by DATA-xPUNGED** [comment]
> I dunno if you're joking but that is a picture of you seeing nothing.

love how many people just genuinely cannot tell the difference between night vision and not using it and others see it clear as day

**Comment by Iamgoofball** [comment]
without night vision:
dreamseeker_JFDzXc4e06
with night vision:
dreamseeker_8pSVzoaQOi

**Comment by SomeoneYouProbablyKnow** [comment]
That is the same picture. You cannot tell me otherwise. Night vision is apparently a psy op

**Comment by optimumtact** [comment]
if you see the difference here it's because your gamma is turned way up because you're a powergamer.

**Comment by SimplySmiley123** [comment]
Goof, these pictures are the same thing.
Checked my gamma and messed around with it for a bit, there is no difference.

**Comment by nianjiilical** [comment]
the difference is noticeable to me but im on a hdr monitor

**Comment by optimumtact** [comment]
it's only noticeable if you darken your room and shutout external lights reflecting on the monitor.

**Comment by Iamgoofball** [comment]
maybe this is just the ultimate filter of poor monitors vs good monitors

**Comment by DATA-xPUNGED** [comment]
I can see it clearly both on my phone and on my PC monitor, maybe it's not about the monitors but who is looking at it

**Comment by SomeoneYouProbablyKnow** [comment]
I think touching grass in the last 24 hours prevents you from telling them apart

**Comment by Admiral-Gigaton** [comment]
> Highest picked positive quirks last quarter: Friendly (23.1k) (+4) Bilingual (18.3k) (+4) Light Step (15.6k) (+8) Musician (12.5k) (+2) Signer (10.8k) (+2)

why does no one pick jolly?? its so good!

**Comment by KingkumaArt** [comment]
> Also, it’d be nice to have a 0 point quirk that does what old light drinker did. I had it on because slurred speech is funny but carrying industrial amounts of alcohol and killing your liver for a joke is annoying. 

This right here is why I run it regardless of if I need points.
I wish I had a photo of the time I lept on the bar with my gripper gloves and smashed my bottle trying to set it down, leaving a drunk secoff laying facedown in a pool of broken glass and space beer, slurring various swearwords at nobody in particular.

**Comment by SomeoneYouProbablyKnow** [comment]
> why does no one pick jolly?? its so good!

Because most people optimize for power instead of for effort which is why you see em spend 20 minutes each round grinding for an item that's gonna give em 0.05 % better idds of living 5 minutes more. But don't snitch on Jolly like that, no one likes engaging with mood and jolly lets you circumvent it with minimal effort

**Comment by necromanceranne** [comment]
> > Highest picked positive quirks last quarter: Friendly (23.1k) (+4) Bilingual (18.3k) (+4) Light Step (15.6k) (+8) Musician (12.5k) (+2) Signer (10.8k) (+2)
> 
> why does no one pick jolly?? its so good!

nobody on this station is happy to be here

**Comment by breadcrombs** [comment]
too early for April Fools

**Comment by Exester509** [comment]
> Goof, these pictures are the same thing. Checked my gamma and messed around with it for a bit, there is no difference.

I think you can see the wires leading up to the power terminal and to the airlock in the second one.
Or at least they're easier to notice for me

**Comment by SimplySmiley123** [comment]
> > Goof, these pictures are the same thing. Checked my gamma and messed around with it for a bit, there is no difference.
> 
> I think you can see the wires leading up to the power terminal and to the airlock in the second one. Or at least they're easier to notice for me

Nope. Stared at it for a while off and on. Can't tell any difference between those images.

**Comment by RedBaronFlyer** [comment]
I can only really tell a difference when I have my window curtains up, lights turned off, and my monitor brightness turned up to perpetual flashbang level brightness.

**Comment by Iamgoofball** [comment]
i can see it clearly in a bright well lit room with sunlight so idk whats up with u gamers

**Comment by SimplySmiley123** [comment]
> i can see it clearly in a bright well lit room with sunlight so idk whats up with u gamers

Dunno, could say the same for you since you're the one seeing things who also wants to replace something with the utterly horrid to use nightmare vision.

**Comment by rellyreri** [comment]
guys please calibrate your monitors properly

**Comment by RedBaronFlyer** [comment]
> guys please calibrate your monitors properly

no

**Comment by Dorsisdwarf** [comment]
SS13 crew are already total lightweights, they have one small glass of beer and are immediately hiccuping, slurring, and giggling for 5 minutes with an obnoxious echo. and a blur filter.

**Comment by Dawnseer** [comment]
I believe the title is a bit at odds with the content of the pr. Making quirks objectively worse for the sake of making them, well, just worse, is not making them more interesting.

**Comment by kawoppi** [comment]
Light drinker is still avoidable by just not drinking any alcohol. But I'm not sure that's something that necessarily has to be solved. Light drinker is currently a fun quirk for when you join some people at the bar and it says something about the character. One gripe with it is that by the time you can get blackout drunk you'll be dying of liver failure so that fun is over before it can even begin. Having it make you go blackout drunk immediately fixes this, but it also removes all of the buildup and will probably discourage some from drinking anything in the first place.

**Comment by Admiral-Gigaton** [comment]
> Light drinker is still avoidable by just not drinking any alcohol. But I'm not sure that's something that necessarily has to be solved. Light drinker is currently a fun quirk for when you join some people at the bar and it says something about the character. One gripe with it is that by the time you can get blackout drunk you'll be dying of liver failure so that fun is over before it can even begin. Having it make you go blackout drunk immediately fixes this, but it also removes all of the buildup and will probably discourage some from drinking anything in the first place.

I sure do love it when someone gets a singular unit of booze in me, and then some ghost possesses me and griefs me, or that person griefs me. love doing that, cant wait for a 2 point trait to be worse than a 6 pointer

**Comment by SomeoneYouProbablyKnow** [comment]
Where is the shadowy cabal forcing you to take light drinker? And more importantly what are you using the points on? Positive quirks are all ... underwhelming, by design

**Comment by Dawnseer** [comment]
And so negative quirks should be overwhelming? Funny.

File: code/datums/mood.dm [hunk 2]
```diff
@@ -18,8 +18,11 @@
 	var/sanity = SANITY_NEUTRAL
 	/// the total combined value of all visible moodlets for the mob
 	var/shown_mood
-	/// Moodlet value modifier
-	var/mood_modifier = 1
+	/// Moodlet value modifiers
+	var/mood_modifier = list(
+		MOOD_MODIFIER_GOOD = 1,
+		MOOD_MODIFIER_BAD = 1,
+	)
```
Comment (MrMelbert) [inline code/datums/mood.dm:25]: This probably doesn't need to be a list. just a `var/good_mood_modifier` and a `var/bad_mood_modifier`. It's not like there's any other moodlet categories than "good" and "bad".

**Review by MrMelbert — CHANGES_REQUESTED** [review]
No to light drinker in its current state, might be funnier to just crank up the current modifier to like 20 instead of 1.25. 
Or give them an alternate liver that's very vulnerable to alcohol or something. 

No to nightmare vision in its current state, the effect is just obnoxious (and this is coming from someone who regularly unironically uses the goggles when they find it). Maybe if it was changed to a different effect. 

Might be cooking with hypersensitive though.

File: code/datums/mood.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/datums/mood.dm:25]: i think that's hardcoding for the sake of hardcoding, what if someone has another category of moodlets they wanna fuck w/ down the line?

**Comment by jughu** [comment]
just remove night vision completely

**Comment by DATA-xPUNGED** [comment]
> just remove night vision completely

This might honestly be for the best. Having a quirk that only has a noticeable effect on CERTAIN monitors and IRL lightning conditions kinda sucks

**Comment by optimumtact** [comment]
i dont even know why this is still open there's no appetite for changes.

**Pull request closed without merging**